### PR TITLE
Add keyspace meta in codec

### PR DESCRIPTION
**Pull request by ystaticy**

### CODE REVIEW

**Review by disksing — APPROVED** [review]

File: internal/apicodec/codec_v2.go [hunk 1]
```diff
@@ -49,26 +50,27 @@ func BuildKeyspaceName(name string) string {
...
+	keyspaceMeta *keyspacepb.KeyspaceMeta
 }
 
 // NewCodecV2 returns a codec that can be used to encode/decode
 // keys and requests to and from APIv2 format.
-func NewCodecV2(mode Mode, keyspaceID uint32) (Codec, error) {
-	if keyspaceID > maxKeyspaceID {
+func NewCodecV2(mode Mode, keyspaceMeta *keyspacepb.KeyspaceMeta) (Codec, error) {
+	keyspaceID := keyspaceMeta.Id
+	if keyspaceMeta.Id > maxKeyspaceID {
```
Comment (iosmanthus) [inline internal/apicodec/codec_v2.go:63]:
```suggestion
	if keyspaceID > maxKeyspaceID {
```

**Review by AmoebaProtozoa — APPROVED** [review]

File: internal/apicodec/codec_v2_test.go [hunk 2]
```diff
@@ -110,62 +114,63 @@ func (suite *testCodecV2Suite) TestNewCodecV2() {
 	re := suite.Require()
 	testCases := []struct {
 		mode           Mode
-		spaceID        uint32
+		keyspaceMeta   keyspacepb.KeyspaceMeta
```
Comment (iosmanthus) [inline internal/apicodec/codec_v2_test.go:117]: How about keeping the spaceID untouched, and constructing the keyspaceMeta via the spaceID.

File: internal/apicodec/codec_v2_test.go [hunk 3]
```diff
@@ -110,62 +114,63 @@ func (suite *testCodecV2Suite) TestNewCodecV2() {
...
-			spaceID:        1<<24 - 1,
+			keyspaceMeta:   keyspacepb.KeyspaceMeta{Id: 1<<24 - 1},
 			expectedPrefix: []byte{'r', 255, 255, 255},
 			expectedEnd:    []byte{'s', 0, 0, 0},
 		},
 	}
 	for _, testCase := range testCases {
 		if testCase.shouldErr {
-			_, err := NewCodecV2(testCase.mode, testCase.spaceID)
+			_, err := NewCodecV2(testCase.mode, &testCase.keyspaceMeta)
```
Comment (iosmanthus) [inline internal/apicodec/codec_v2_test.go:164]:
```suggestion
			_, err := NewCodecV2(testCase.mode, &keyspacepb.KeyspaceMeta{Id: spaceID})
```

File: internal/locate/pd_codec.go [hunk 4]
```diff
@@ -85,6 +85,15 @@ func GetKeyspaceID(client pd.Client, name string) (uint32, error) {
 	return meta.Id, nil
 }
 
+// GetKeyspaceMeta attempts to retrieve keyspace meta corresponding to the given keyspace name from PD.
+func GetKeyspaceMeta(client pd.Client, name string) (*keyspacepb.KeyspaceMeta, error) {
```
Comment (iosmanthus) [inline internal/locate/pd_codec.go:89]: If `GetKeyspaceID` is still used in some external package, we might implement it using `GetKeyspaceMeta`.

**Comment by iosmanthus** [comment]
Rest LGTM! ♥️

**Pull request merged**